authentication/pipeline/user.py

```diff
@@ -211,3 +216,28 @@ def activate_user(
         user.save()
 
     return {}
+
+
+def create_courseware_user(
+    strategy, backend, user=None, is_new=False, **kwargs
+):
+    """
+    Create a user in the courseware, deferring a retry via celery if it fails
+
+    Args:
+        user (users.models.User): the user that was just created
+        is_new (bool): True if the user was just created
+    """
+    if not is_new:
+        return {}
+
+    try:
+        courseware_api.create_user(user)
+    except Exception:  # pylint: disable=broad-except
+        log.exception("Error creating courseware user records on User create")
+        # try again later
+        courseware_tasks.create_user_from_id.apply_async(
+            (user.id,), countdown=CREATE_COURSEWARE_USER_RETRY_DELAY
```

Comment:
Needs to be defined (formerly in `signals.py`)

Reply:
Might or might not be my instance config again?  I updated `honor_code` in `REGISTRATION_EXTRA_FIELDS`

```
web_1     | [2019-06-21 18:53:22] WARNING 22 [django.request] log.py:228 - [114be66b94cd] - Bad Request: /oauth2/authorize/
web_1     | [2019-06-21 18:53:22] WARNING 22 [django.request] log.py:228 - [114be66b94cd] - Bad Request: /oauth2/authorize/
nginx_1   | 172.21.0.5 - - [21/Jun/2019:18:53:22 +0000] "GET /oauth2/authorize/?scope=user%3Aread&state=pZZBZtpzZ6vzJn8fKqbeBUNl3KR8N7GH&redirect_uri=http%3A%2F%2Fhost.docker.internal%3A18000%2Fauth%2Fcomplete%2Fmitxpro-oauth2%2F%3Fredirect_state%3DpZZBZtpzZ6vzJn8fKqbeBUNl3KR8N7GH&response_type=code&client_id=AjkqATCHMEnyn6IDuK3xrt5pxMFVR68pfzRhYp0T HTTP/1.1" 400 1337 "-" "python-requests/2.21.0" "-"
web_1     | [pid: 22|app: 0|req: 8/12] 172.21.0.5 () {38 vars in 1200 bytes} [Fri Jun 21 18:53:22 2019] GET /oauth2/authorize/?scope=user%3Aread&state=pZZBZtpzZ6vzJn8fKqbeBUNl3KR8N7GH&redirect_uri=http%3A%2F%2Fhost.docker.internal%3A18000%2Fauth%2Fcomplete%2Fmitxpro-oauth2%2F%3Fredirect_state%3DpZZBZtpzZ6vzJn8fKqbeBUNl3KR8N7GH&response_type=code&client_id=AjkqATCHMEnyn6IDuK3xrt5pxMFVR68pfzRhYp0T => generated 1337 bytes in 37 msecs (HTTP/1.1 400) 4 headers in 148 bytes (1 switches on core 0)
web_1     | [2019-06-21 18:53:22] ERROR 22 [root] user.py:237 - [114be66b94cd] - Error creating courseware user records on User create
web_1     | Traceback (most recent call last):
web_1     |   File "./authentication/pipeline/user.py", line 235, in create_courseware_user
web_1     |     courseware_api.create_user(user)
web_1     |   File "./courseware/api.py", line 50, in create_user
web_1     |     create_edx_auth_token(user)
web_1     |   File "/usr/local/lib/python3.6/contextlib.py", line 52, in inner
web_1     |     return func(*args, **kwds)
web_1     |   File "./courseware/api.py", line 147, in create_edx_auth_token
web_1     |     resp.raise_for_status()
web_1     |   File "/usr/local/lib/python3.6/site-packages/requests/models.py", line 940, in raise_for_status
web_1     |     raise HTTPError(http_error_msg, response=self)
web_1     | requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: http://xpro.odl.local:8053/oauth2/authorize/?scope=user%3Aread&state=pZZBZtpzZ6vzJn8fKqbeBUNl3KR8N7GH&redirect_uri=http%3A%2F%2Fhost.docker.internal%3A18000%2Fauth%2Fcomplete%2Fmitxpro-oauth2%2F%3Fredirect_state%3DpZZBZtpzZ6vzJn8fKqbeBUNl3KR8N7GH&response_type=code&client_id=AjkqATCHMEnyn6IDuK3xrt5pxMFVR68pfzRhYp0T
```

Reply:
That is likely a secret or client id mismatch

Reply:
or possibly a redirect_uri issue

Reply:
Besides the error probably caused by some misconfiguration on my end, it does wait to trigger the edx user creation until after I complete filling out all details and runs the sync task a minute later.

Reply:
Error was caused by me wiping out my db an hour earlier and forgetting to update oauth application info. Works great now.